src/ploomber/telemetry/telemetry.py

```diff
@@ -107,39 +114,41 @@ def get_env():
         return 'local'
 
 
-# Checks if a specific directory/file exists, creates if not
-# In case the user didn't set a custom dir, will turn to the default home
-def check_dir_file_exist(name=None, is_file=None):
+def check_dir_exist(name=None):
+    """
+    Checks if a specific directory exists, creates if not.
+    In case the user didn't set a custom dir, will turn to the default home
+    """
     if PLOOMBER_HOME_DIR:
-        final_location = os.path.expanduser(PLOOMBER_HOME_DIR)
+        final_location = PLOOMBER_HOME_DIR
     else:
-        final_location = os.path.expanduser(DEFAULT_HOME_DIR)
+        final_location = DEFAULT_HOME_DIR
 
     if name:
-        final_location = os.path.join(final_location, name)
+        final_location = f"{final_location}/{name}"
+    p = Path(final_location)
+    if not p.exists():
+        p.mkdir(parents=True)
 
-    if not os.path.exists(final_location):
-        if is_file:
-            os.mknod(final_location)
-        else:
-            os.makedirs(final_location)
     return final_location
 
 
 def check_uid():
-    # Check if local telemetry exists as a uid
-    uid_path = os.path.join(check_dir_file_exist(CONF_DIR), 'uid.yaml')
-    if not os.path.exists(uid_path):  # Create - doesn't exist
+    """
+    Checks if local user id exists as a uid file, creates if not.
+    """
+    uid_path = Path(check_dir_exist(CONF_DIR), 'uid.yaml')
+    if not uid_path.exists():  # Create - doesn't exist
         uid = str(uuid.uuid4())
         try:  # Create for future runs
-            with open(uid_path, "w") as file:
+            with uid_path.open("w", encoding="utf-8") as file:
```

Comment:
can we leave the default encoding instead of hardcoding? this is system-dependent. no need to change it but for future reference you can use `Path.write_text` and `Path.read_text` instead of `Path.open`

Reply:
Sure, where I can I'll try these functions, here I figured it's better to enforce a yaml structure via the class, without open it will not work. You can take a look of this thread: https://stackoverflow.com/questions/57296168/pathlib-path-write-text-in-append-mode